### PR TITLE
util.svg tagged template

### PR DESCRIPTION
**Pull request by Geliogabalus**

### CODE REVIEW

File: src/util/svgTagTemplate.mjs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            markupNode.selector = selectorAttribute.value;
+            attributes.removeNamedItem('@selector');
+        } else {
+            markupNode.selector = tagName;
+        }
+
+        const groupSelectorAttribute = attributes.getNamedItem('@group-selector');
+        if (groupSelectorAttribute) {
+            const groupSelectors = groupSelectorAttribute.value.split(',');
+            if (groupSelectors.length === 1) {
```
Comment (kumilingus) [inline src/util/svgTagTemplate.mjs:44]: I don't think we need to branch it here. This code will work fine with one and more group selectors.
```
markupNode.groupSelector = groupSelectors.map(s => s.trim())
```

File: docs/src/joint/api/dia/Cell/markup.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <pre class="docs-method-signature"><code>markup</code></pre>
 
 <p>
-    Either an XML string or JSON markup specifying an array of JSON elements.
+    Either an XML string or JSON markup specifying an array of JSON elements. Also JSON markup can be described using <a href="#util.svg">svg</a> tagged template.
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/dia/Cell/markup.html:4]: "...using **an** <a href..."

File: docs/src/joint/api/dia/Cell/markup.html [hunk 3]
```diff
@@ -67,29 +67,63 @@ <h4 id="dia.Cell.markup.json">JSON Markup</h4>
     }]
 }]</code></pre>
 
+<h4 id="dia.Cell.markup.json">JSON Markup using tagged template</h4>
+
+<p>
+    JSON markup also can be defined using <a href="#util.svg">svg</a> tagged template. This tagged template converts SVG representation of the markup into JSON markup object.
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/dia/Cell/markup.html:73]: "...using **an** <a href..."

"into **a** JSON markup ..."

File: docs/src/joint/api/dia/Cell/markup.html [hunk 4]
```diff
@@ -67,29 +67,63 @@ <h4 id="dia.Cell.markup.json">JSON Markup</h4>
...
+    Let's write previous example using svg tagged template:
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/dia/Cell/markup.html:74]: "...write **the** previous example using **an** svg ...."

File: docs/src/joint/api/util/svg.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<pre class="docs-method-signature"><code>svg(strings)</code></pre>
+<p>
+    This is the tagged template which converts string into a <a href="#dia.Cell.markup">markup</a> object during declaring a <a href="#dia.Cell">Cell</a>.
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/util/svg.html:3]: "...converts **a** string"

"...object **while** declaring..."

File: docs/src/joint/api/util/svg.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    The resulting object contains fields which are descripted in <a href="#dia.Cell.markup.json">this part of the documentation.</a>
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/util/svg.html:4]: "...which are **described** in..."

File: docs/src/joint/api/util/svg.html [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            <td>selector</td>
+            <td><code>@selector</code> attribute.</td>
+        </tr>
+        <tr>
+            <td>groupSelector</td>
+            <td><code>@group-selector</code> attribute. Accepts comma-separated lists e.g. <code>@group-selector="group1, group2"</code>.</td>
+        </tr>
+        <tr>
+            <td>namespaceURI</td>
+            <td>The namespace URI of the element. It defaults to SVG namespace <a target="_blank" href="https://www.w3.org/2000/svg"><code>"http://www.w3.org/2000/svg"</code></a>.</td>
```
Comment (jamesgeorgewilliams) [inline docs/src/joint/api/util/svg.html:25]: "...to **the** SVG namespace..."

**Pull request merged**